evcache-core/src/main/java/net/spy/memcached/EVCacheMemcachedClient.java

```diff
@@ -255,7 +258,12 @@ public void complete() {
             }
 
             public void gotData(String k, int flags, long cas, byte[] data) {
-                if (!key.equals(k)) log.warn("Wrong key returned. Key - " + key + "; Returned Key " + k);
+                if (!key.equals(k)) {
```

Comment:
Note that this is the only place we don't return. I'm wondering if that was deliberate or not.